### PR TITLE
add nil check in kubelet

### PR DESCRIPTION
**Pull request by wanlonghenry**
add nil check in kubelet to address this issue: https://msazure.visualstudio.com/InfrastructureInsights/_workitems/edit/16519364

### CODE REVIEW

File: source/plugins/ruby/kubelet_utils.rb [hunk 1]
```diff
@@ -56,9 +56,9 @@ def get_node_allocatable(cpu_capacity, memory_capacity)
         parsed_response = JSON.parse(allocatable_response.body)
 
         begin
-          kubereserved_cpu = parsed_response["kubeletconfig"]["kubeReserved"]["cpu"]
-          if kubereserved_cpu.nil? || kubereserved_cpu == ""
-            kubereserved_cpu = "0.0"
+          kubereserved_cpu = "0.0"
+          if parsed_response.key?("kubeletconfig") && parsed_response["kubeletconfig"].key?("kubeReserved") && parsed_response["kubeletconfig"]["kubeReserved"].key?("cpu")
```
Comment (jatakiajanvi12) [inline source/plugins/ruby/kubelet_utils.rb:60]: You haven't added the empty string check: kubereserved_cpu == "". The keys can be present but the value can still be empty right?

Same everywhere else

**Review by jatakiajanvi12 — CHANGES_REQUESTED** [review]

File: source/plugins/ruby/kubelet_utils.rb (hunk 1 above)
Comment (wanlonghenry) [reply source/plugins/ruby/kubelet_utils.rb:60]: from the previous logic, if the kubereserved_cpu == "" then the value will be overriding to 0.

Comment (wanlonghenry) [reply source/plugins/ruby/kubelet_utils.rb]: addressed

**Review by wanlonghenry — COMMENTED** [review]
changed

**Review by ganga1980 — APPROVED** [review]

**Review by jatakiajanvi12 — APPROVED** [review]

**Pull request merged**